src/loc.rs

```diff
@@ -0,0 +1,466 @@
+        }
+
+        let input = self.debug_loc_section.range_from(offset..);
+        Ok(LocationListIter::new(input, address_size, base_address))
+    }
+
+    /// Iterate over the raw `LocationListEntry`s starting at the given offset.
+    ///
+    /// The `address_size` must be match the compilation unit for this location list.
+    ///
+    /// This iterator does not perform any processing of the location entries,
+    /// such as handling base addresses.
+    ///
+    /// Can be [used with
+    /// `FallibleIterator`](./index.html#using-with-fallibleiterator).
+    pub fn raw_locations(&self,
+                         offset: DebugLocOffset,
+                         address_size: u8)
+                         -> Result<RawLocationListIter<Endian>> {
+        let offset = offset.0 as usize;
+        if self.debug_loc_section.len() < offset {
+            return Err(Error::UnexpectedEof);
+        }
+
+        let input = self.debug_loc_section.range_from(offset..);
+        Ok(RawLocationListIter::new(input, address_size))
+    }
+}
+
+/// A raw iterator over a location list.
+///
+/// This iterator does not perform any processing of the location entries,
+/// such as handling base addresses.
+#[derive(Debug)]
+pub struct RawLocationListIter<'input, Endian>
+    where Endian: Endianity
+{
+    input: EndianBuf<'input, Endian>,
+    address_size: u8,
+}
+
+impl<'input, Endian> RawLocationListIter<'input, Endian>
+    where Endian: Endianity
+{
+    /// Construct a `RawLocationListIter`.
+    pub fn new(input: EndianBuf<'input, Endian>,
+               address_size: u8)
+               -> RawLocationListIter<'input, Endian> {
+        RawLocationListIter {
+            input: input,
+            address_size: address_size,
+        }
+    }
+
+    /// Advance the iterator to the next location.
+    pub fn next(&mut self) -> Result<Option<LocationListEntry<'input, Endian>>> {
+        if self.input.is_empty() {
+            return Ok(None);
+        }
+
+        let (rest, location) = try!(LocationListEntry::parse(self.input, self.address_size));
+        if location.range.is_end() {
+            self.input = EndianBuf::new(&[]);
+        } else {
+            self.input = rest;
+        }
+
+        Ok(Some(location))
+    }
+}
+
+impl<'input, Endian> FallibleIterator for RawLocationListIter<'input, Endian>
+    where Endian: Endianity
+{
+    type Item = LocationListEntry<'input, Endian>;
+    type Error = Error;
+
+    fn next(&mut self) -> ::std::result::Result<Option<Self::Item>, Self::Error> {
+        RawLocationListIter::next(self)
+    }
+}
+
+/// An iterator over a location list.
+#[derive(Debug)]
+pub struct LocationListIter<'input, Endian>
+    where Endian: Endianity
+{
+    raw: RawLocationListIter<'input, Endian>,
+    base_address: u64,
+}
+
+impl<'input, Endian> LocationListIter<'input, Endian>
+    where Endian: Endianity
+{
+    /// Construct a `LocationListIter`.
+    pub fn new(input: EndianBuf<'input, Endian>,
```

Comment:
Should this be `pub`? A lot of our iterators don't have public constructors, but its something that I've always been on the fence with. On the one hand, it means people can only get valid iterators (barring bugs in our library) but on the other hand it makes mocking more difficult.